### PR TITLE
Update for new cosmos-sdk

### PR DESCRIPTION
**Pull request by mappum**
> Thank you a lot for contributing to Cosmos Voyager!

### Issue

closes: #588

### Screenshots

❤️ Thank you!

### CODE REVIEW

**Comment by faboweb** [comment]
Great! Thanks for the initiative! :)

**Comment by mappum** [comment]
This is basically ready for the demo, now I'm just trying to update the tests. For some reason the LCD process isn't getting started in the main process spec 🤔

**Comment by nylira** [comment]
I think if tests are all that are standing in the way of the demo, we should merge this and finish the tests later.

**Comment by faboweb** [comment]
Checking it out now

**Comment by faboweb** [comment]
merged and fixed main process initialization

**Comment by nylira** [comment]
I think I've fixed all of the unit test issues -- it works locally. But CircleCI is showing an error still.

**Comment by mappum** [comment]
This PR is theoretically ready, just can't get the tests to pass on Circle (they pass locally).

**Comment by nylira** [comment]
CircleCI's configuration [1] is still downloading an old `gaia 0.5.0` binary, when we should be downloading a `basecoind 0.13.0 797aec2` [2] instead. 

[1] https://github.com/cosmos/voyager/blob/d5a9a9c86ae262735f508ea79120d08170bc608e/.circleci/config.yml#L37 
[2] https://github.com/cosmos/voyager/pull/591/files#diff-1aef8e5caa4eaa5b1857799628184f1bR1

Greg isn't around to build and host `basecoind` for us, so we'll have to do it ourselves temporarily.

I don't know if this is related to the `./app/unit/specs/main.spec.js` test timing out.

**Comment by faboweb** [comment]
unit tests shouldn't be affected by the binaries.
could catch @nylira with the binary version

**Comment by faboweb** [comment]
Amazing work! Thank you for fixing this. But would this also work with the e2e tests reenabled?

We can also push this to develop and add a follow up to reenable e2e tests.

**Comment by mappum** [comment]
@NodeGuy wow 😲 Thank you! That green checkmark is the light at the end of a very long, dark tunnel.

@faboweb I think there are some things to update in the e2e tests for these changes.

**Comment by mappum** [comment]
>We can also push this to develop and add a follow up to reenable e2e tests.

IMO we should do this to un-bottleneck the team, I can do the work on the e2e tests.

**Comment by nylira** [comment]
I didn't get a chance to run this locally yet (with Cosmos SDK 0.13), and I'm out of time for today. Can someone else try it?

**Comment by faboweb** [comment]
On it 👍

File: app/src/main/index.js [hunk 1]
```diff
@@ -212,35 +213,36 @@ async function startLCD(home, nodeIP) {
...
       // '--trust-node'
     ])
     logProcess(child, join(home, "lcd.log"))
 
-    // XXX why the hell stderr?!?!?!?
-    child.stderr.on("data", data => {
-      if (data.includes("Serving on")) resolve(child)
-    })
+    // XXX: should wait for server to be ready
+    setTimeout(() => resolve(child), 1000)
```
[the remark this answers is not in the document]
Comment (mappum) [reply app/src/main/index.js]: It no longer logs anything about starting the server, we could have a loop that polls until the request goes through.

File: app/src/main/index.js [hunk 2]
```diff
@@ -588,34 +589,45 @@ async function main() {
...
   if (seeds.length === 0) {
     throw new Error("No seeds specified in config.toml")
   }
 
   // choose one random node to start from
   nodeIP = pickNode(seeds)
 
-  let _lcdInitialized = await lcdInitialized(join(root, "lcd"))
-  log("LCD is" + (_lcdInitialized ? "" : " not") + " initialized")
+  let _lcdInitialized = true // await lcdInitialized(join(root, 'lcd'))
```
[the remark this answers is not in the document]
Comment (mappum) [reply app/src/main/index.js]: Yes, we will re-enable it once the SDK team implements `init` again.

File: app/src/renderer/vuex/modules/send.js [hunk 3]
```diff
@@ -1,114 +1,66 @@
...
-            reject(err)
-            if (args.reject) args.reject(err)
-          } else {
-            resolve(res)
-            if (args.resolve) args.resolve(res)
-          }
-        }
+    // wait to ensure tx is committed before we query
+    // XXX
+    setTimeout(() => dispatch("queryWalletBalances"), 3 * 1000)
```
[the remark this answers is not in the document]
Comment (mappum) [reply app/src/renderer/vuex/modules/send.js]: Not sure how to do it cleanly, might need to listen for new blocks via RPC (which isn't verified by the LCD right now).

[the remark this answers is not in the document]
Comment (faboweb) [reply app/src/renderer/vuex/modules/send.js]: Ok, could you maybe create a follow up issue so we don't forget to implement that in the future?

File: app/src/main/index.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (faboweb) [reply app/src/main/index.js]: I added an issue so we reimplement that in the SDK: https://github.com/cosmos/cosmos-sdk/issues/973

File: app/src/renderer/vuex/modules/send.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (mappum) [reply app/src/renderer/vuex/modules/send.js]: Opened #679

**Comment by nylira** [comment]
Amazing work, everyone!

**Pull request merged**